### PR TITLE
[8.0] porting of ricevute bancarie v2

### PR DESCRIPTION
**Pull request by scigghia**
this is a refactory of PR https://github.com/OCA/l10n-italy/pull/83

### CODE REVIEW

**Comment by eLBati** [comment]
Thanks @scigghia 
travis is red. Please see https://travis-ci.org/OCA/l10n-italy/builds/66846963

Nice to have: squash 2bb0f3792081254344c8090bf94a0ddd3cc73cad and 9a4d53495c01b2d9580b7e20c19f83453843e77e

**Comment by eLBati** [comment]
Other open point: ```riba.configurazione``` and ```riba.distinta*``` models

**Comment by scigghia** [comment]
here my porting of l10n_it_ricevute_bancarie for Odoo
I need someone to test it, from the begin to the end.
for me, it works
obviously it is not entirely ported to new v8 api, we can do that after it works for others, not only for me ;)

**Comment by eLBati** [comment]
Restarted hanged travis build.

Runbot complains about
```
2015-06-17 13:41:43	WARNING	server	openerp.modules.loading:365 load_modules The model riba.configuration has no access rules, consider adding one. E.g. access_riba_configuration,access_riba_configuration,model_riba_configuration,,1,0,0,0
2015-06-17 13:41:43	WARNING	server	openerp.modules.loading:365 load_modules The model riba.list has no access rules, consider adding one. E.g. access_riba_list,access_riba_list,model_riba_list,,1,0,0,0
2015-06-17 13:41:43	WARNING	server	openerp.modules.loading:365 load_modules The model riba.list.line has no access rules, consider adding one. E.g. access_riba_list_line,access_riba_list_line,model_riba_list_line,,1,0,0,0
2015-06-17 13:41:43	WARNING	server	openerp.modules.loading:365 load_modules The model riba.list.move.line has no access rules, consider adding one. E.g. access_riba_list_move_line,access_riba_list_move_line,model_riba_list_move_line,,1,0,0,0
```
please fix it.

About ```riba.configurazione``` and ```riba.distinta*``` models, if you want to keep the new names, migration script should be provided.
However, don't you think we would better keep the old names?

**Comment by tafaRU** [comment]
Needs a rebase.

**Comment by scigghia** [comment]
@tafaRU rebase done

**Comment by tafaRU** [comment]
@scigghia, there are still conflicts that must be resolved.

File: l10n_it_ricevute_bancarie/__openerp__.py [hunk 1]
```diff
@@ -27,46 +27,73 @@
...
+Ad ogni passaggio di stato sarà possibile generare le relative registrazioni
+contabili, le quali verranno riepilogate nel tab 'contabilità'.
+Questo tab è presente sia sulla distinta che sulle sue righe.
 
-Qui https://docs.google.com/document/d/1xCqeTcY6CF-Dgk_Avthhy7iwg_aG86WzNv3E_HHQkt4/edit# abbiamo un esempio delle tipiche registrazioni generate da un flusso 'salvo buon fine'.
+Qui
+http://goo.gl/jpRhJp abbiamo un esempio delle tipiche registrazioni generate
+da un flusso 'salvo buon fine'.
     """,
     'images': [],
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/__openerp__.py]: Remove empty key

File: l10n_it_ricevute_bancarie/models/riba.py [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import fields, models, api, _, exceptions, workflow
+# import time
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/models/riba.py]: Remove useless commented line

**Comment by scigghia** [comment]
confilcts solved

**Comment by scigghia** [comment]
can be merged?

**Comment by tafaRU** [comment]
@scigghia please see errors at https://travis-ci.org/OCA/l10n-italy/jobs/92969922

**Comment by eLBati** [comment]
@scigghia see https://github.com/scigghia/l10n-italy/pull/5

**Comment by scigghia** [comment]
is it mergeable?

**Comment by eLBati** [comment]
:+1:

**Comment by andrea4ever** [comment]
:+1:

**Comment by OpenCode** [comment]
:+1:

File: l10n_it_ricevute_bancarie/__openerp__.py [hunk 3]
```diff
@@ -26,47 +26,36 @@
...
-I possibili stati della distinta sono: bozza, accettata, accreditata, pagata, insoluta, annullata.
-Ad ogni passaggio di stato sarà possibile generare le relative registrazioni contabili, le quali verranno riepilogate nel tab 'contabilità'. Questo tab è presente sia sulla distinta che sulle sue righe.
-
-Qui https://docs.google.com/document/d/1xCqeTcY6CF-Dgk_Avthhy7iwg_aG86WzNv3E_HHQkt4/edit# abbiamo un esempio delle tipiche registrazioni generate da un flusso 'salvo buon fine'.
-    """,
-    'images': [],
-    'depends': ['account','account_voucher', 'l10n_it_fiscalcode', 'account_due_list'],
-    'init_xml': [],
+    'depends': [
+        'account',
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/__openerp__.py]: account seems redundant, isn'it?

File: l10n_it_ricevute_bancarie/models/riba.py [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+        move_lines = self.env['account.move.line']
+        for line in self.line_ids:
+            move_lines |= line.payment_ids
+        self.payment_ids = move_lines
+
+    _name = 'riba.distinta'
+    _description = 'Riba list'
+
+    name = fields.Char(
+        'Reference', size=128, required=True, readonly=True,
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/models/riba.py]: is there any good reason to set a size limit there?

File: l10n_it_ricevute_bancarie/models/riba.py [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+        required=True,
+        default=lambda self: fields.Date.context_today(self),
+        help="Keep empty to use the current date")
+
+    @api.multi
+    def unlink(self):
+        for riba_list in self:
+            if riba_list.state not in ('draft',  'cancel'):
+                raise exceptions.Warning(
+                    _('Error'),
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/models/riba.py]: No title in Warning

File: l10n_it_ricevute_bancarie/views/account_config_view.xml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+                            </div>
+                        </div>
+                </group>
+            </xpath>
+        </field>
+    </record>
+    
+
+</data>
+</openerp>
```
Comment (tafaRU) [inline l10n_it_ricevute_bancarie/views/account_config_view.xml]: No newline at the end of file

**Comment by tafaRU** [comment]
@scigghia thanks for the contrib!
Few remarks otherwise :+1:

File: l10n_it_ricevute_bancarie/__openerp__.py (hunk 3 above)
Comment (scigghia) [inline l10n_it_ricevute_bancarie/__openerp__.py]: yes

File: l10n_it_ricevute_bancarie/models/riba.py (hunk 4 above)
Comment (scigghia) [inline l10n_it_ricevute_bancarie/models/riba.py]: no

File: l10n_it_ricevute_bancarie/models/riba.py (hunk 5 above)
Comment (scigghia) [inline l10n_it_ricevute_bancarie/models/riba.py]: i'm changing to UserError

File: l10n_it_ricevute_bancarie/views/account_config_view.xml (hunk 6 above)
Comment (scigghia) [inline l10n_it_ricevute_bancarie/views/account_config_view.xml]: ops!

**Comment by lepistone** [comment]
@scigghia red because of lint

**Comment by scigghia** [comment]
i'm going to fix

**Comment by scigghia** [comment]
merge?

**Comment by eLBati** [comment]
As travis is green, I merge

**Comment by scigghia** [comment]
champagne! :)

**Pull request merged**